java/src/main/java/ai/rapids/cudf/ColumnView.java

```diff
@@ -673,8 +673,13 @@ public final ColumnVector[] slice(int... indices) {
         nativeHandles[i] = 0;
       }
     } catch (Throwable t) {
-      cleanupColumnViews(nativeHandles, columnVectors);
-      throw t;
+      try {
+        cleanupColumnViews(nativeHandles, columnVectors, t);
+      } catch (Throwable s) {
+        t.addSuppressed(s);
+      } finally {
+        throw t;
+      }
```

Comment:
Nit: There's no need for a finally block here, it's already catching everything.  Throwing from a `finally` which is normally not desired, since it has the smashing effect I mentioned earlier, so could change to this here and elsewhere this pattern appears.
```suggestion
      } catch (Throwable s) {
        t.addSuppressed(s);
      }
      throw t;
```